src/transpile/fusion.hpp

```diff
@@ -299,28 +304,60 @@ op_t Fusion::generate_fusion_operation(const std::vector<op_t>& fusioned_ops,
   ExperimentData dummy_data;
 
   if (method == Method::unitary) {
-    // Unitary simulation
-    QubitUnitary::State<> unitary_simulator;
-    unitary_simulator.initialize_qreg(qubits.size());
-    unitary_simulator.apply_ops(fusioned_ops, dummy_data, dummy_rng);
-    return Operations::make_unitary(qubits, unitary_simulator.qreg().move_to_matrix(),
-                                    std::string("fusion"));
+#if defined(_MSC_VER) || defined(GNUC_AVX2)
+    if (!disable_simd_) {
+      // Unitary simulation
+      QubitUnitary::State<QV::UnitaryMatrixAvx2<>> unitary_simulator;
+      unitary_simulator.initialize_qreg(qubits.size());
+      unitary_simulator.apply_ops(fusioned_ops, dummy_data, dummy_rng);
+      return Operations::make_unitary(qubits, unitary_simulator.qreg().move_to_matrix(),
+                                      std::string("fusion"));
+    } else
+#endif
```

Comment:
This ifdef block is wrong, since if it is not used it will run this simulation twice. But i think it should just be removed for a normal if / else block like the controller classes